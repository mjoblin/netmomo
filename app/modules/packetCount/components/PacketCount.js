import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { VictoryChart, VictoryLine, VictoryTheme } from 'victory';
import moment from 'moment';


class PacketCount extends React.Component {
    constructor(...args) {
        super(...args);

        this.firstTime = true;

        this.state = {
            data: [{when: moment(), y: 0}]
        };
    }

    componentWillReceiveProps(nextProps) {
        if (this.firstTime) {
            this.setState({
                data: [
                    {when: moment(), y: nextProps.packetCount.Ethernet}
                ]
            });
            this.firstTime = false;
        }

        let data2 = this.state.data;
        data2 = data2.slice(-5);

        this.setState({
            data: [
                ...data2,
                {when: moment(), y: nextProps.packetCount.Ethernet}
            ]
        })
    }

    render() {
        const packetCount = this.props.packetCount;

        return (
            <div>
                <div>
                    <h2>Packet counts</h2>
                    <table>
                        {
                            Object.keys(packetCount).map(protocol => (
                                <tr key={protocol}>
                                    <td>{protocol}</td>
                                    <td>{packetCount[protocol].toLocaleString()}</td>
                                </tr>
                            ))
                        }
                    </table>
                </div>
                <div>
                    <VictoryChart
                        theme={VictoryTheme.material}
                        width={1400}
                        height={800}
                        padding={100}
                        animate={{duration: 500, onLoad: {duration: 1000}}}
                        easing="linear"
                    >
                        <VictoryLine
                            interpolation="linear"
                            data={this.state.data}
                            x={d => moment(d.when).format('hh:mm:ss')}
                            scale="log"
                        />
                    </VictoryChart>
                </div>
            </div>
        );
    }
}


PacketCount.propTypes = {
    packetCount: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
    packetCount: state.packetCount,
});

export default connect(
    mapStateToProps,
)(PacketCount);

