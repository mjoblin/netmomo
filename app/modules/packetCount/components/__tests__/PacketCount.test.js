import React from 'react';
import { shallow } from 'enzyme';
import { PacketCount } from '../PacketCount';
import testData from './testData.json';

describe('PacketCount', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <PacketCount
                packetCounts={testData}
            />
        );

        expect(component).toMatchSnapshot();
    });
});