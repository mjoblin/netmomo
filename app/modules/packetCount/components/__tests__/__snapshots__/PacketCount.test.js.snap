// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacketCount matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PacketCount
    packetCounts={
      Array [
        Object {
          "count": 94410,
          "key": "Ethernet",
          "protocol": "Ethernet",
        },
        Object {
          "count": 93325,
          "key": "IP",
          "protocol": "IP",
        },
        Object {
          "count": 94243,
          "key": "UDP",
          "protocol": "UDP",
        },
        Object {
          "count": 94289,
          "key": "Raw",
          "protocol": "Raw",
        },
        Object {
          "count": 987,
          "key": "IPv6",
          "protocol": "IPv6",
        },
        Object {
          "count": 98,
          "key": "ARP",
          "protocol": "ARP",
        },
        Object {
          "count": 17,
          "key": "Padding",
          "protocol": "Padding",
        },
        Object {
          "count": 69,
          "key": "TCP",
          "protocol": "TCP",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Packet counts
        </h2>,
        <Tabs
          animated={false}
          defaultActiveKey="packets_table"
          hideAdd={false}
          prefixCls="ant-tabs"
        >
          <TabPane
            placeholder={null}
            tab="Table"
          >
            <PacketCountTable
              packetCounts={
                Array [
                  Object {
                    "count": 94410,
                    "key": "Ethernet",
                    "protocol": "Ethernet",
                  },
                  Object {
                    "count": 93325,
                    "key": "IP",
                    "protocol": "IP",
                  },
                  Object {
                    "count": 94243,
                    "key": "UDP",
                    "protocol": "UDP",
                  },
                  Object {
                    "count": 94289,
                    "key": "Raw",
                    "protocol": "Raw",
                  },
                  Object {
                    "count": 987,
                    "key": "IPv6",
                    "protocol": "IPv6",
                  },
                  Object {
                    "count": 98,
                    "key": "ARP",
                    "protocol": "ARP",
                  },
                  Object {
                    "count": 17,
                    "key": "Padding",
                    "protocol": "Padding",
                  },
                  Object {
                    "count": 69,
                    "key": "TCP",
                    "protocol": "TCP",
                  },
                ]
              }
            />
          </TabPane>
          <TabPane
            placeholder={null}
            tab="Chart"
          >
            <PacketCountChart
              data={Array []}
            />
          </TabPane>
        </Tabs>,
      ],
      "className": "packet-count",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Packet counts",
        },
        "ref": null,
        "rendered": "Packet counts",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "children": Array [
            <TabPane
              placeholder={null}
              tab="Table"
            >
              <PacketCountTable
                packetCounts={
                  Array [
                    Object {
                      "count": 94410,
                      "key": "Ethernet",
                      "protocol": "Ethernet",
                    },
                    Object {
                      "count": 93325,
                      "key": "IP",
                      "protocol": "IP",
                    },
                    Object {
                      "count": 94243,
                      "key": "UDP",
                      "protocol": "UDP",
                    },
                    Object {
                      "count": 94289,
                      "key": "Raw",
                      "protocol": "Raw",
                    },
                    Object {
                      "count": 987,
                      "key": "IPv6",
                      "protocol": "IPv6",
                    },
                    Object {
                      "count": 98,
                      "key": "ARP",
                      "protocol": "ARP",
                    },
                    Object {
                      "count": 17,
                      "key": "Padding",
                      "protocol": "Padding",
                    },
                    Object {
                      "count": 69,
                      "key": "TCP",
                      "protocol": "TCP",
                    },
                  ]
                }
              />
            </TabPane>,
            <TabPane
              placeholder={null}
              tab="Chart"
            >
              <PacketCountChart
                data={Array []}
              />
            </TabPane>,
          ],
          "defaultActiveKey": "packets_table",
          "hideAdd": false,
          "prefixCls": "ant-tabs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "packets_table",
            "nodeType": "class",
            "props": Object {
              "children": <PacketCountTable
                packetCounts={
                  Array [
                    Object {
                      "count": 94410,
                      "key": "Ethernet",
                      "protocol": "Ethernet",
                    },
                    Object {
                      "count": 93325,
                      "key": "IP",
                      "protocol": "IP",
                    },
                    Object {
                      "count": 94243,
                      "key": "UDP",
                      "protocol": "UDP",
                    },
                    Object {
                      "count": 94289,
                      "key": "Raw",
                      "protocol": "Raw",
                    },
                    Object {
                      "count": 987,
                      "key": "IPv6",
                      "protocol": "IPv6",
                    },
                    Object {
                      "count": 98,
                      "key": "ARP",
                      "protocol": "ARP",
                    },
                    Object {
                      "count": 17,
                      "key": "Padding",
                      "protocol": "Padding",
                    },
                    Object {
                      "count": 69,
                      "key": "TCP",
                      "protocol": "TCP",
                    },
                  ]
                }
              />,
              "placeholder": null,
              "tab": "Table",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "packetCounts": Array [
                  Object {
                    "count": 94410,
                    "key": "Ethernet",
                    "protocol": "Ethernet",
                  },
                  Object {
                    "count": 93325,
                    "key": "IP",
                    "protocol": "IP",
                  },
                  Object {
                    "count": 94243,
                    "key": "UDP",
                    "protocol": "UDP",
                  },
                  Object {
                    "count": 94289,
                    "key": "Raw",
                    "protocol": "Raw",
                  },
                  Object {
                    "count": 987,
                    "key": "IPv6",
                    "protocol": "IPv6",
                  },
                  Object {
                    "count": 98,
                    "key": "ARP",
                    "protocol": "ARP",
                  },
                  Object {
                    "count": 17,
                    "key": "Padding",
                    "protocol": "Padding",
                  },
                  Object {
                    "count": 69,
                    "key": "TCP",
                    "protocol": "TCP",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "packets_chart",
            "nodeType": "class",
            "props": Object {
              "children": <PacketCountChart
                data={Array []}
              />,
              "placeholder": null,
              "tab": "Chart",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "data": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Packet counts
          </h2>,
          <Tabs
            animated={false}
            defaultActiveKey="packets_table"
            hideAdd={false}
            prefixCls="ant-tabs"
          >
            <TabPane
              placeholder={null}
              tab="Table"
            >
              <PacketCountTable
                packetCounts={
                  Array [
                    Object {
                      "count": 94410,
                      "key": "Ethernet",
                      "protocol": "Ethernet",
                    },
                    Object {
                      "count": 93325,
                      "key": "IP",
                      "protocol": "IP",
                    },
                    Object {
                      "count": 94243,
                      "key": "UDP",
                      "protocol": "UDP",
                    },
                    Object {
                      "count": 94289,
                      "key": "Raw",
                      "protocol": "Raw",
                    },
                    Object {
                      "count": 987,
                      "key": "IPv6",
                      "protocol": "IPv6",
                    },
                    Object {
                      "count": 98,
                      "key": "ARP",
                      "protocol": "ARP",
                    },
                    Object {
                      "count": 17,
                      "key": "Padding",
                      "protocol": "Padding",
                    },
                    Object {
                      "count": 69,
                      "key": "TCP",
                      "protocol": "TCP",
                    },
                  ]
                }
              />
            </TabPane>
            <TabPane
              placeholder={null}
              tab="Chart"
            >
              <PacketCountChart
                data={Array []}
              />
            </TabPane>
          </Tabs>,
        ],
        "className": "packet-count",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Packet counts",
          },
          "ref": null,
          "rendered": "Packet counts",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "children": Array [
              <TabPane
                placeholder={null}
                tab="Table"
              >
                <PacketCountTable
                  packetCounts={
                    Array [
                      Object {
                        "count": 94410,
                        "key": "Ethernet",
                        "protocol": "Ethernet",
                      },
                      Object {
                        "count": 93325,
                        "key": "IP",
                        "protocol": "IP",
                      },
                      Object {
                        "count": 94243,
                        "key": "UDP",
                        "protocol": "UDP",
                      },
                      Object {
                        "count": 94289,
                        "key": "Raw",
                        "protocol": "Raw",
                      },
                      Object {
                        "count": 987,
                        "key": "IPv6",
                        "protocol": "IPv6",
                      },
                      Object {
                        "count": 98,
                        "key": "ARP",
                        "protocol": "ARP",
                      },
                      Object {
                        "count": 17,
                        "key": "Padding",
                        "protocol": "Padding",
                      },
                      Object {
                        "count": 69,
                        "key": "TCP",
                        "protocol": "TCP",
                      },
                    ]
                  }
                />
              </TabPane>,
              <TabPane
                placeholder={null}
                tab="Chart"
              >
                <PacketCountChart
                  data={Array []}
                />
              </TabPane>,
            ],
            "defaultActiveKey": "packets_table",
            "hideAdd": false,
            "prefixCls": "ant-tabs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "packets_table",
              "nodeType": "class",
              "props": Object {
                "children": <PacketCountTable
                  packetCounts={
                    Array [
                      Object {
                        "count": 94410,
                        "key": "Ethernet",
                        "protocol": "Ethernet",
                      },
                      Object {
                        "count": 93325,
                        "key": "IP",
                        "protocol": "IP",
                      },
                      Object {
                        "count": 94243,
                        "key": "UDP",
                        "protocol": "UDP",
                      },
                      Object {
                        "count": 94289,
                        "key": "Raw",
                        "protocol": "Raw",
                      },
                      Object {
                        "count": 987,
                        "key": "IPv6",
                        "protocol": "IPv6",
                      },
                      Object {
                        "count": 98,
                        "key": "ARP",
                        "protocol": "ARP",
                      },
                      Object {
                        "count": 17,
                        "key": "Padding",
                        "protocol": "Padding",
                      },
                      Object {
                        "count": 69,
                        "key": "TCP",
                        "protocol": "TCP",
                      },
                    ]
                  }
                />,
                "placeholder": null,
                "tab": "Table",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "packetCounts": Array [
                    Object {
                      "count": 94410,
                      "key": "Ethernet",
                      "protocol": "Ethernet",
                    },
                    Object {
                      "count": 93325,
                      "key": "IP",
                      "protocol": "IP",
                    },
                    Object {
                      "count": 94243,
                      "key": "UDP",
                      "protocol": "UDP",
                    },
                    Object {
                      "count": 94289,
                      "key": "Raw",
                      "protocol": "Raw",
                    },
                    Object {
                      "count": 987,
                      "key": "IPv6",
                      "protocol": "IPv6",
                    },
                    Object {
                      "count": 98,
                      "key": "ARP",
                      "protocol": "ARP",
                    },
                    Object {
                      "count": 17,
                      "key": "Padding",
                      "protocol": "Padding",
                    },
                    Object {
                      "count": 69,
                      "key": "TCP",
                      "protocol": "TCP",
                    },
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "packets_chart",
              "nodeType": "class",
              "props": Object {
                "children": <PacketCountChart
                  data={Array []}
                />,
                "placeholder": null,
                "tab": "Chart",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "data": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
